### PR TITLE
changelog v0.13

### PR DESCRIPTION
**Pull request by jasonwilliams**
in progress

Blog post:
https://github.com/boa-dev/boa-dev.github.io/pull/14

## Notes

* Special thanks @neeldug for keeping on top of our CI! Making sure our Github actions and testing were working properly.
* We also had some Rust updates that needed us to [refactor some parts of our code](https://github.com/boa-dev/boa/pull/1352)
* There has been more work on the VM, in fact it has been [re-designed](https://github.com/boa-dev/boa/pull/1361) to be more efficient
* Worked on some builtins being more [spec compliant](https://github.com/boa-dev/boa/pull/1422)
* A [better design](https://github.com/boa-dev/boa/pull/1432) of property descriptors that follow the spec more closely

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
 # CHANGELOG
 
-# 0.13
+# 0.13.0 (2021-09-30)
 
 Feature Enhancements:
 
+- [FEATURE #1526](https://github.com/boa-dev/boa/pull/1526) :Implement ComputedPropertyName for accessor properties in ObjectLiteral (@raskad)
```
Comment (Razican) [inline CHANGELOG.md:7]:
```suggestion
- [FEATURE #1526](https://github.com/boa-dev/boa/pull/1526): Implement ComputedPropertyName for accessor properties in ObjectLiteral (@raskad)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +25,11 @@ Feature Enhancements:
 
 Bug Fixes:
 
+- [BUG #152](https://github.com/boa-dev/boa/pull/1521): Added "js" feature for getrandom for WebAssembly builds (@Razican)
```
Comment (Razican) [inline CHANGELOG.md:28]:
```suggestion
- [BUG #1521](https://github.com/boa-dev/boa/pull/1521): Added "js" feature for getrandom for WebAssembly builds (@Razican)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,95 @@
...
+Feature Enhancements:
+
+- [FEATURE #1526](https://github.com/boa-dev/boa/pull/1526): Implement ComputedPropertyName for accessor properties in ObjectLiteral (@raskad)
+- [FEATURE #1365](https://github.com/boa-dev/boa/pull/1365): Implement splice method (@neeldug)
+- [FEATURE #1364](https://github.com/boa-dev/boa/pull/1364): Implement spread for objects (@FrancisMurillo)
+- [FEATURE #1525](https://github.com/boa-dev/boa/pull/1525): Implement Object.preventExtensions() and Object.isExtensible() (@HalidOdat)
+- [FEATURE #1508](https://github.com/boa-dev/boa/pull/1508): Implement Object.values() (@HalidOdat)
+- [FEATURE #1332](https://github.com/boa-dev/boa/pull/1332): Implement Array.prototype.sort (@jedel1043)
+- [FEATURE #1417](https://github.com/boa-dev/boa/pull/1471): Implement Object.keys and Object.entries (@skyne98)
+- [FEATURE #1406](https://github.com/boa-dev/boa/pull/1406): Implement destructuring assignments (@raskad)
```
Comment (Razican) [inline CHANGELOG.md:14]: I would say this could be added to the blog post :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,95 @@
...
+- [FEATURE #1469](https://github.com/boa-dev/boa/pull/1469): Implement String.prototype.replaceAll (@raskad)
+- [FEATURE #1442](https://github.com/boa-dev/boa/pull/1442): Implement closure functions (@HalidOdat)
```
Comment (Razican) [inline CHANGELOG.md:16]: We should mention in the blog post these improvements to the API

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,95 @@
...
+- [BUG #1349](https://github.com/boa-dev/boa/pull/1349): Fixes Array.protoype.includes (@neeldug)
+- [BUG #1348](https://github.com/boa-dev/boa/pull/1348): Fixes unshift maximum size (@neeldug)
+- [BUG #1339](https://github.com/boa-dev/boa/pull/1339): Scripts should not be considered in a block (@macmv)
+- [BUG #1312](https://github.com/boa-dev/boa/pull/1312): Fix display for nodes (@macmv)
+- [BUG #1347](https://github.com/boa-dev/boa/pull/1347): Fix stringpad abstract operation (@neeldug)
+
+Internal Improvements:
+
+- [INTERNAL #1584](https://github.com/boa-dev/boa/pull/1584): Refactor the Math builtin object (spec compliant) (@jedel1043)
+- [INTERNAL #1523](https://github.com/boa-dev/boa/pull/1523): Allow moving NativeObject variables into closures as external captures (@jedel1043)
```
Comment (Razican) [inline CHANGELOG.md:49]: I would say this is a feature, since it makes the API more usable and is visible from outside.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,95 @@
...
+- [INTERNAL #1535](https://github.com/boa-dev/boa/pull/1535): Refactor JSON.parse (@raskad)
+- [INTERNAL #1569](https://github.com/boa-dev/boa/pull/1569): Refactor EnvironmentRecordTrait functions (@raskad)
+- [INTERNAL #1572](https://github.com/boa-dev/boa/pull/1572): Refactor builtin Map intrinsics to follow more closely the spec (@jedel1043)
+- [INTERNAL #1445](https://github.com/boa-dev/boa/pull/1445): improve map conformance without losing perf (@neeldug)
```
Comment (Razican) [inline CHANGELOG.md:53]: Some spec improvements are considered fixes but some others not. What should we do?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,95 @@
 # CHANGELOG
 
+# 0.13.0 (2021-09-30)
```
Comment (Razican) [inline CHANGELOG.md:3]: We should probably add a subtitle and a link to the changes here, as with 0.12.

**Review by Razican — COMMENTED** [review]
Looking good!

File: CHANGELOG.md (hunk 4 above)
Comment (raskad) [reply CHANGELOG.md:16]: Yes I think for this in combination with #1523 we could show a little example in the blog post, possibly just from the https://github.com/boa-dev/boa/blob/master/boa/examples/closures.rs file.

File: CHANGELOG.md (hunk 6 above)
Comment (raskad) [reply CHANGELOG.md:53]: I feel like we should bump some of the internals to fixes and features. @Razican already called out #1523 as a feature. I looked at all the internals and I think these ones could qualify fixes, as they mostly only improve spec compliance without changing too much of the internals: #1584, #1535, #1572, #1445, #1488, #1463, #1434, #1350, #1331

**Pull request merged**